subprojects/core/src/test/groovy/org/gradle/api/internal/DefaultNamedDomainObjectSetSpec.groovy

```diff
@@ -356,6 +356,28 @@ class DefaultNamedDomainObjectSetSpec extends AbstractNamedDomainObjectCollectio
         }
     }
 
+    def "name based filtering does not realize pending"() {
```

Comment:
🤔 I think a test that assert the behavior when doing `container.toList()` or `container.all {}` is missing